irc_client/config.py

```diff
@@ -206,9 +208,7 @@ def _validate(self, *junk: object) -> bool:
             return False
         # TODO: can realname be empty?
 
-        from .backend import NICK_REGEX, CHANNEL_REGEX
-
-        if not re.fullmatch(NICK_REGEX, self._nick_entry.get()):
+        if not re.fullmatch(backend.NICK_REGEX, self._nick_entry.get()):
```

Comment:
I don't like the changes in this file. Avoiding the import cycle is quite fragile.